lib/improver/utilities/cube_manipulation_new.py

```diff
@@ -312,363 +314,372 @@ def concatenate_cubes(
+            cube.add_aux_coord(new_model_id_coord)
+            cube.add_aux_coord(new_model_coord)
 
-    Raises:
-        ValueError:
-            If bounds ranges at different points are incompatible
-    """
-    for name in coord_list:
-        coord = cube.coord(name)
-        if coord.bounds is None:
-            continue
-        if len(coord.points) == 1:
-            continue
-
-        bounds_ranges = np.abs(np.diff(coord.bounds))
-        reference_range = bounds_ranges[0]
-        if not np.all(np.isclose(bounds_ranges, reference_range)):
-            msg = ('Cube with mismatching {} bounds ranges '
-                   'cannot be blended'.format(name))
-            raise ValueError(msg)
+    def process(self, cubes_in, blend_coord, cycletime=None,
+                weighting_coord=None):
+        """
+        Prepares merged input cube for cycle and grid blending
 
+        Args:
+            cubes (iris.cube.CubeList or iris.cube.Cube):
+                Cubes to be merged.
+            blend_coord (str):
+                Name of coordinate over which blending will be performed.
 
-def _equalise_cell_methods(cubes):
-    """
-    Function to equalise cell methods that do not match.
+        Kwargs:
+            cycletime (str or None):
+                The cycletime in a YYYYMMDDTHHMMZ format e.g. 20171122T0100Z.
+                Required for rationalise_blend_time_coordinates.
+            weighting_coord (str or None):
+                The coordinate across which weights will be scaled in a
+                multi-model blend.  Required for
+                rationalise_blend_time_coordinates.
 
-    Args:
-        cubes (iris.cube.CubeList):
-            List of cubes to check the cell methods and revise.
-    Returns:
-        cubelist (iris.cube.CubeList):
-            List of cubes with revised cell methods.
-            Currently the cell methods are simply deleted if
-            they do not match.
-    Warns:
-        Warning: If only a single cube.
+        Returns:
+            iris.cube.Cube:
+                Merged cube.
 
-    """
-    if len(cubes) == 1:
-        msg = ('Only a single cube so no differences will be found '
-               'in cell methods')
-        warnings.warn(msg)
-        cubelist = cubes
-    else:
-        cell_methods = cubes[0].cell_methods
-        for cube in cubes[1:]:
-            cell_methods = list(set(cell_methods) & set(cube.cell_methods))
-        cubelist = cubes
-        for cube in cubelist:
-            cube.cell_methods = tuple(cell_methods)
-    return cubelist
+        Raises:
+            ValueError:
+                If trying to blend over model when model_id_attr is not set
+        """
+        # if input is already a single cube, return unchanged
+        if isinstance(cubes_in, iris.cube.Cube):
+            return cubes_in
+
+        # test for model blend
+        if "model" in blend_coord and self.model_id_attr is None:
```

Comment:
This was deliberate.  I rather like the idea of this class being opaque - the user can specify "model" as the blend coordinate, and the specific names / nature of the `model_id` and `model_configuration` coordinate is known only within the blending code.  This gives us flexibility to change how we handle this metadata without affecting the user interface.  We can also then get the if / else statement out of the CLI and stop worrying about if the CLI argument is `model_id`, `model_configuration` or just `model`.

What do you think?

Reply:
(Note the use of `if "model" in blend_coord` is also consistent with what we're doing in `rationalise_blend_time_coords`.)

Reply:
Do you think that it is worth adding something to the `blend_coord` docstring (`Name of coordinate over which blending will be performed.`)? Something like `Name of coordinate over which blending will be performed. If blending models, then a 'model' coordinate is supported.` This is just because `model` is not actually the name of any coordinate, despite the docstring saying that `blend_coord` should be a `name of a coordinate`.

Reply:
Updated docstring to explain this.